src/contracts/registry.ts

```diff
@@ -0,0 +1,68 @@
+import { getContractAddress } from "./contract";
+import { HexString } from "../types/Strings";
+import { getConfig } from "../config/config";
+import { ContractTransaction } from "ethers";
+import { MissingContract, MissingProvider, MissingSigner } from "../utilities";
+import { Registry__factory } from "../types/typechain";
+
+const CONTRACT_NAME = "Registry";
+
+/**
+ * Get the JSON RPC error from the body, if one exists
+ * @param e The error expected to have a vm Error
+ *
+ * @returns the error if any
+ */
+const getVmError = (e: { body?: string }): string | undefined => {
+  try {
+    const parsed = JSON.parse(e.body || "{}");
+    return parsed?.error?.message;
+  } catch (e) {
+    return undefined;
+  }
+};
+
+/**
+ * resolveHandleToId() Try to resolve a handle into a DSNP Id
+ *
+ * @param handle String handle to resolve
+ * @returns The Hex for the DSNP Id or null if not found
+ */
+export const resolveHandleToId = async (handle: string): Promise<HexString | null> => {
+  const contract = await getContract();
+  try {
+    return (await contract.resolveHandleToId(handle)).toHexString();
+  } catch (e) {
+    const vmError = getVmError(e);
+    if (vmError === "Error: VM Exception while processing transaction: revert Handle does not exist") {
+      return null;
+    }
+    throw e;
+  }
+};
+
+/**
+ * register() registers a handle to get a new DSNP Id
+ *
+ * @param identityContractAddress Address of the identity contract to use
+ * @param handle The string handle to register
+ * @returns The contract Transaction
+ */
+export const register = async (identityContractAddress: HexString, handle: string): Promise<ContractTransaction> => {
```

Comment:
question: do users have the ability to set default identityContractAddress in the config?

Reply:
I wonder if I should change this name. This is the full identity contract or the proxy contract, not the logic contract. So this is unique to the user.

Reply:
Ohhh ok I see.